app/assets/javascripts/sitewide/form_cleanups.js

```diff
@@ -0,0 +1,38 @@
+$.fn.clearFormErrors = function () {
+  $(this).find('.nav.nav-tabs li').removeClass('has-error');
+  $(this).find('.form-group').removeClass('has-error');
+  $(this).find('span.help-block').remove();
+};
+
+$.fn.clearFormFields = function () {
+  $(this).find("input")
+    .not("button")
+    .not('input[type="submit"], input[type="reset"], input[type="hidden"]')
+    .not('input[type="radio"]') // Leave out radios as this messes up Bootstrap btn-groups
+    .val('')
+    .removeAttr('checked')
+    .removeAttr('selected');
+};
+
+$.fn.removeBlankFileForms = function () {
+  $(this).find("input[type='file']").each(function () {
+    if (!this.files[0]) {
+      $(this).closest("fieldset").remove();
+    }
+  });
+}
+
+/*
+ * Not the actual Excel tables, but are similar.
+ */
+$.fn.removeBlankExcelTables = function (editMode) {
```

Comment:
Not true, I've added this BECAUSE blank tables don't get saved on server-side, hence I concluded this is expected behavior, so I just added the client-side code also (to reduce requests made to the server, as it is with other validators). Go ahead, comment out call to this function in `steps.js` and see it for yourself. I dare you...